src/Manager/FbcManager.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusFacebookPlugin\Manager;
+
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * We need this manager to be able to access fbc
+ * at different stages of Request and generate it only once
+ */
+final class FbcManager implements FbcManagerInterface
+{
+    private RequestStack $requestStack;
+
+    private int $fbcTtl;
+
+    private string $fbcCookieName;
+
+    /**
+     * Cached value of fbc generated from fbclid at current request
+     */
+    private ?string $generatedFbc = null;
+
+    public function __construct(
+        RequestStack $requestStack,
+        int $fbcTtl,
+        string $fbcCookieName = 'ssfp_fbc'
```

Comment:
```suggestion
        string $fbcCookieName = 'ssf_fbc'
```
I don't use the 'plugin' part in acronyms elsewhere